power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/common_solver_functions.hpp

```diff
@@ -167,6 +167,7 @@ inline void calculate_load_gen_result(IdxRange const& load_gens, Idx bus_number,
         }
         output.load_gen[load_gen].i = conj(output.load_gen[load_gen].s / output.u[bus_number]);
     }
+    (void)std::forward<LoadGenFunc>(load_gen_func);
```

Comment:
why this?

Reply:
this function captures `func` by perfect forwarding. It is not allowed by the standard to not forward a forwarded argument, or move a moved argument.

Reply:
This and the other for file `common_solver_functions.hpp`, the root cause is we define the argument for the lambda as `LoadGenFunc&& load_gen_func`, which is not correct. 

Lambdas in function argument should be passed by value unless there is a strong reason of not to do.

If we change that to `LoadGenFunc load_gen_func`. The stuff will go away.

Reply:
good point. i'll double-check wherever this is used

Reply:
https://github.com/PowerGridModel/power-grid-model/blob/main/power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/common_solver_functions.hpp#L175-L191 `calculate_load_gen_result` is in the hot-loop. should be captured by reference there

Reply:
Pass by value is the standard practice for lambdas as a function argument. The lambda object itself is not so big.  Note that the internal state of the lambda is already captured by reference.

Also, in Release build the compiler might inline all the functions in one go including the lambda self.

Reply:
while standard practice, what's the reason you so strongly object preventing passing by perfect forwarding? as the person writing the function, you by design don't know whether the person calling the function at a later state will capture by reference, or do a single expensive copy. perfect forwarding handles at the function-creation-side, not as a user responsibility. This makes it much more fail-safe on both ends.

Reply:
Same goes with all STL algorithms, they take lambda as value. STL authors just assume that the user of STL algorithm will create cheap-to-copy lambda.

Reply:
done